src/net_processing.cpp

```diff
@@ -2853,7 +2853,9 @@ static bool SendRejectsAndCheckIfBanned(CNode* pnode, CConnman* connman)
                 LogPrintf("Warning: not banning local peer %s!\n", pnode->addr.ToString());
             else
             {
-                connman->Ban(pnode->addr, BanReasonNodeMisbehaving);
+                if (m_banman) {
```

Comment:
Commit [banman: create and split out banman].

I feel like this should be checked outside the surrounding `if (pnode->addr.IsLocal())` statement. Doesn't make sense to log "not banning local peer" if it wouldn't have been banned anyway.

Reply:
good point, will fix.